File: Java/src/main/java/com/gildedrose/item/Backstage.java
package com.gildedrose.item;

public class Backstage extends Item {

    public static String BACKSTAGE = "Backstage passes to a TAFKAL80ETC concert";

    public Backstage(int sellIn, int quality) {
        super(BACKSTAGE, sellIn, quality);
    }

    @Override
    public void increaseQuality() {
        super.increaseQuality();
        if (this.sellIn < 11 && this.quality < 50) {
            super.increaseQuality();
        }
        if (this.sellIn < 6 && this.quality < 50) {
            super.increaseQuality();
        }
    }

    @Override
    public void updateQuality() {
        if (this.quality < 50) {
            this.increaseQuality();
        }
        if (this.sellIn < 0) {
            this.quality = 0;
        }
    }
}

File: python/backstage.py
from item_wrappers import ItemWrapper

class BackstagePassItem(ItemWrapper):
    def update(self):
        self.increase_quality()

        if self.item.sell_in <= 10:
            self.increase_quality()
        if self.item.sell_in <= 5:
            self.increase_quality()

        self.item.sell_in -= 1

        if self.item.sell_in < 0:
            self.item.quality = 0

    def increase_quality(self):
        if self.item.quality < 50:
            self.item.quality += 1
